src/nginx.go

```diff
@@ -1,41 +1,92 @@
 package main
 
 import (
+	"fmt"
 	sdk_args "github.com/newrelic/infra-integrations-sdk/args"
+	"github.com/newrelic/infra-integrations-sdk/data/metric"
+	"github.com/newrelic/infra-integrations-sdk/integration"
 	"github.com/newrelic/infra-integrations-sdk/log"
-	"github.com/newrelic/infra-integrations-sdk/sdk"
+	"net/url"
 )
 
 type argumentList struct {
 	sdk_args.DefaultArgumentList
-	StatusURL  string `default:"http://127.0.0.1/status" help:"NGINX status URL."`
-	ConfigPath string `default:"/etc/nginx/nginx.conf" help:"NGINX configuration file."`
+	StatusURL        string `default:"http://127.0.0.1/status" help:"NGINX status URL."`
+	ConfigPath       string `default:"/etc/nginx/nginx.conf" help:"NGINX configuration file."`
+	RemoteMonitoring bool   `default:"false" help:"Identifies the monitored entity as 'remote'. In doubt: set to true."`
 }
 
 const (
 	integrationName    = "com.newrelic.nginx"
-	integrationVersion = "1.0.2"
+	integrationVersion = "1.1.0"
+	entityRemoteType   = "nginx"
+
+	httpsProtocol    = `https`
+	httpDefaultPort  = `80`
+	httpsDefaultPort = `443`
 )
 
 var (
 	args argumentList
 )
 
 func main() {
-	integration, err := sdk.NewIntegration(integrationName, integrationVersion, &args)
+	i, err := integration.New(integrationName, integrationVersion, integration.Args(&args))
 	fatalIfErr(err)
 	log.SetupLogging(args.Verbose)
 
-	if args.All || args.Inventory {
-		fatalIfErr(setInventoryData(integration.Inventory))
+	hostname, port, err := parseStatusURL(args.StatusURL)
```

Comment:
This comment is related to backwards compatibility: What happen if args.StatusURL doesn't have a Hostname, but `RemoteMonitoring` is false?

In this case will return an error, but since `status URL` is not used in local entity should be this the case?